### ISSUE TITLE
Should icon reflect current state or action?

### ISSUE THREAD

**Issue by sbatten** [microsoft/vscode#131545]
Testing #131196

The locked icon is shown unlocked currently when the current group is locked. I am always confused by whether an icon shows the current status or the status after clicking the button (i.e. the action to take). This is because we are not consistent. It seems we tend to favor action over the status but there are exceptions. 

*SCM: Show as Tree/List*
Shows the action, not the current state
![image](https://user-images.githubusercontent.com/6561887/130664594-0d7bc331-a1ca-4f70-abaf-9fdf0d7a97f3.png)

*Search: Collapse/Expand All*
Shows the action, not the current state

*Explorer: Filter*
Shows the current state, not the action
![image](https://user-images.githubusercontent.com/6561887/130664974-f4a54382-bda6-4621-a304-02d3062640b3.png)

*Timeline: Pinned*
Shows the current state, not the action
![image](https://user-images.githubusercontent.com/6561887/130665068-f03b1c36-d001-423d-8567-33dff0959a34.png)

*GitLens Remotes: Show as Tree/List*
Shows the current state, not the action
![image](https://user-images.githubusercontent.com/6561887/130665315-c693eea7-1836-497a-aeae-a9bd664f3245.png)

*GitLens Search Commits: Keep Results*
Shows the current state, not the action
![image](https://user-images.githubusercontent.com/6561887/130665719-3c3b03d2-ba5d-4d66-b547-e041f74259a3.png)

Can we establish some guideline here? Maybe cool for the editor lock would be showing as locked until hover indicating that click would unlock the group.

**Comment by bpasero** [comment]
Good catch, I tend to just flip the icon to show the lock because we do the same for pinned tabs:

![image](https://user-images.githubusercontent.com/900690/130734294-edd6f741-4037-47c4-9460-a9e4f0cc1fe2.png)

//cc @misolori for the general UX guidance
